src/Symfony/Component/AssetMapper/Compiler/JavaScriptImportPathCompiler.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\Compiler;
+
+use Symfony\Component\AssetMapper\AssetMapper;
+use Symfony\Component\AssetMapper\MappedAsset;
+
+/**
+ * Resolves import paths in JS files.
+ *
+ * @experimental
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class JavaScriptImportPathCompiler implements AssetCompilerInterface
```

Comment:
Do we still need this ? Can't we rely on the import map for project files too, by having all the available local assets in the import map (when digests are enabled and so changing the file name is needed) ?

Reply:
Hmm. It absolutely IS needed so that we can automatically discover that `app.js` is importing `./other.js`, and so automatically add `/assets/other.js` to the importmap (and also preload it if `app.js` is marked as preloaded).

What is less certain and relates to your [previous question](https://github.com/symfony/symfony/pull/50112#discussion_r1180681205) is whether or not this class should *adjust* the paths - e.g. `import './other'` to `import './other.js'`.

Reply:
Let me answer your question better :p. Yes, we could decide to put EVERY JavaScript file in the "asset mapper" system into the `importmap` instead of actually looking to see which are needed (i.e. which are being imported). That just felt wasteful to me. But looking for the imports does add complexity.